causalml/inference/tree/causal/_builder.pyx

```diff
@@ -58,7 +58,7 @@ cdef class DepthFirstCausalTreeBuilder(TreeBuilder):
         cdef int init_capacity
 
         if tree.max_depth <= 10:
-            init_capacity = (2 ** (tree.max_depth + 1)) - 1
+            init_capacity = int((2 ** (tree.max_depth + 1)) - 1)
```

Comment:
I think the exponentiation results in type conversion to float. This is causing me issues in the installation.

```
 Error compiling Cython file:
      ------------------------------------------------------------
      ...
      
              # Initial capacity
              cdef int init_capacity
      
              if tree.max_depth <= 10:
                  init_capacity = (2 ** (tree.max_depth + 1)) - 1
                                                              ^
      ------------------------------------------------------------
      
      causalml/inference/tree/causal/_builder.pyx:61:56: Cannot assign type 'double' to 'int'
```

Reply:
That's weird though. The rule in Python is that integer exponentiation results in an integer, how is it giving a double here? Does CPython have different rules?